store/src/java/com/zimbra/cs/imap/ImapSessionManager.java

```diff
@@ -113,19 +112,29 @@ protected static ImapSessionManager getInstance() {
         return SINGLETON;
     }
 
+    /**
+     * Computes a key to use for caching the ImapListener `session` in the `sessions` in-memory cache
+     *
+     * @param session ImapListener
+     * @return a cache key to deterministically identify an IMAP folder for an Account
+     */
+    protected String sessionsCacheKey(ImapListener session) {
+        return String.format("%s:%d", session.getAuthenticatedAccountId(), session.getFolderId());
```

Comment:
How does this work for mountpoints and other users folders (under /home)
My intuition is that it doesn't and you end up wiping out entries in the cache for a folder in the authenticated users mailbox with ones that happen to have the same folder Id from the other user's mailbox.